### PR TITLE
feat(sql,identity): update telemetry

### PR DESCRIPTION
**Pull request by xzf0587**
1 add required properties for telemetry event
2 send error event instead of exception when there is an error

### CODE REVIEW

File: packages/fx-core/src/plugins/resource/identity/utils/telemetryUtil.ts [hunk 1]
```diff
@@ -8,34 +8,52 @@ export class TelemetryUtils {
...
         if (!properties) {
             properties = {};
         }
-        properties[Telemetry.component] = Telemetry.telemetryName;
-        TelemetryUtils.ctx.telemetryReporter?.sendTelemetryEvent(eventName, properties, measurements);
+        TelemetryUtils.send(eventName, properties, measurements);
     }
 
-    public static sendErrorEvent(eventName: string,
+    public static sendSuccessEvent(eventName: string,
```
Comment (adashen) [inline packages/fx-core/src/plugins/resource/identity/utils/telemetryUtil.ts:30]: this method seems could be merged with sendEvent?

File: packages/fx-core/src/plugins/resource/sql/constants.ts [hunk 2]
```diff
@@ -57,22 +57,27 @@ export class Constants {
...
+    static systemError = "system";
 
-    static readonly component = "component";
-    static readonly success = "success";
-    static readonly errorType = "error-type";
-    static readonly errorMessage = "error-message";
+    static readonly stage = {
+        preProvision: "pre-provision",
+        provision: "provision",
+        postProvision: "post-provision",
```
Comment (adashen) [inline packages/fx-core/src/plugins/resource/sql/constants.ts:70]: do we have post-provision? Seems no such event in Changyong's doc

File: packages/fx-core/src/plugins/resource/sql/index.ts [hunk 3]
```diff
@@ -11,26 +12,26 @@ export class SqlPlugin implements Plugin {
...
         return await this.sqlImpl.callFunc(func, ctx);
     }
 
     public async getQuestions(stage: Stage, ctx: PluginContext): Promise<Result<QTreeNode | undefined, FxError>> {
-        return this.runWithSqlError(() => this.sqlImpl.getQuestions(stage, ctx), ctx);
+        return this.runWithSqlError(Telemetry.stage.postProvision, () => this.sqlImpl.getQuestions(stage, ctx), ctx);
     }
 
-    private async runWithSqlError(fn: () => Promise<SqlResult>, ctx: PluginContext): Promise<SqlResult> {
+    private async runWithSqlError(stage: string, fn: () => Promise<SqlResult>, ctx: PluginContext): Promise<SqlResult> {
```
Comment (adashen) [inline packages/fx-core/src/plugins/resource/sql/index.ts:34]: do we have -start event?

File: packages/fx-core/src/plugins/resource/sql/constants.ts (hunk 2 above)
Comment (xzf0587) [reply packages/fx-core/src/plugins/resource/sql/constants.ts:70]: pre and post event are included in the events.
image

**Review by adashen — APPROVED** [review]
Please make sure Changyong/Bowen OK with the PR. Some comment about the sendEvent/sendSucessEvent method. But not block the PR. So approve with comments

File: packages/fx-core/src/plugins/resource/identity/utils/telemetryUtil.ts (hunk 1 above)
Comment (xzf0587) [reply packages/fx-core/src/plugins/resource/identity/utils/telemetryUtil.ts]: It is fixed. Thx.

**Review by chagong — APPROVED** [review]

**Pull request merged**